## Find correlation coefficient

### Given x and y and linear regression equation y = b*x + a + e , solve for a and b such that sum(e ^ 2) is minimum

In [1]:
import os
import pandas as pd
import numpy as np

In [2]:
os.chdir("C:/Users/satish/Downloads")

cars = pd.read_csv("cars.csv")

In [3]:
x= cars["Weight"]
y= cars["MPG"]

sum_yi = np.sum(y)
sum_xi = np.sum(x)
sum_xiyi = np.sum(x*y)
sum_xi_squared = np.sum(x**2)
n = len(x)

In [4]:
a  = ((sum_yi * sum_xi_squared) - (sum_xi * sum_xiyi))/(n * sum_xi_squared - (sum_xi**2))
print(a)

46.325305341697934


In [5]:
b = (n * sum_xiyi - (sum_xi * sum_yi))/(n * sum_xi_squared - (sum_xi**2))
print(b)

-0.00781162853863321


### Thus we've found the coefficient a and intercept b using OLS method. For one unit increase in x, there'll be 0.007 units decrease in y. If x is 0, y will be 46.325